staging/src/k8s.io/component-base/logs/kube_features.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logs
+
+import (
+	"k8s.io/component-base/featuregate"
+)
+
+const (
+	// owner: @pohly
+	// kep: http://kep.k8s.io/3077
+	// alpha: v1.24
+	//
+	// Enables looking up a logger from a context.Context instead of using
+	// the global fallback logger and manipulating the logger that is
+	// used by a call chain.
+	ContextualLogging featuregate.Feature = "ContextualLogging"
+)
+
+// FeatureGates consists of logging feature gates. These must be registered in
+// FeatureGate instances that get passed to functions in this package.
+var FeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
+	ContextualLogging: {Default: false, PreRelease: featuregate.Alpha},
+}
+
+// AddFeatureGates adds all feature gates used by this package.
+func AddFeatureGates(mutableFeatureGate featuregate.MutableFeatureGate) error {
+	return mutableFeatureGate.Add(FeatureGates)
```

Comment:
don't pass a shared instance of FeatureGates to the caller (which could mutate it), pass in a fresh instance each time

Reply:
True. I never liked that Go cannot enforce "const". Changed.